### DISCUSSION TITLE
[V5]: implementing socket.io

### DISCUSSION THREAD

**Discussion by autumn-witch** [adonisjs/core#2280] [Help] [answered]
I'm trying to implement sockets through the socket.io article provided by Virk, but it's not working yet and I'm having a lot of trouble figuring out why. So:
I then created a `Services\Ws file`: [https://pastebin.com/YXiUPFji](url).

I then pasted this: [https://pastebin.com/qyRDuYW1](url) in `start/socket.ts`, and this: [https://pastebin.com/CRPrQ1ML](url) in my AppProvider.
And so, when loading my app, I'm expecting to see `0` / `1` /  `2` logged out in the console, but only 0 and 1 are, therefore meaning that my start `Ws.start` method isn't actually executed, but I have no clue why. So help is very welcome!
Thanks in advance

**Comment by RomainLanz** [comment, marked as answer]
Hey! 👋 

This issue has been fixed in our Discord server.
I'm posting this message to help anyone that also has an issue with Socket.IO.

The issue was a mis-match of dependencies between the frontend and the backend.
**They should both have the same major version.**

You cannot install `socket.io` version `2` in your backend and use `socket.io` version `3` in your frontend.
